lib/Models/Definition/upsertModelFromJson.ts

```diff
@@ -71,22 +76,13 @@ export default function upsertModelFromJson(
     }
   }
 
-  if (model === undefined && options.matchByShareKey && json.id !== undefined) {
-    const potentialId = terria.getModelIdByShareKey(json.id);
-    if (potentialId !== undefined) {
-      model = terria.getModelById(BaseModel, potentialId);
-      if (model === undefined) {
-        errors.push(
-          new TerriaError({
-            message: `Failed to get model \`"${potentialId}"\` found using share key \`"${json.id}"\``
-          })
-        );
-      }
-    }
-  }
   if (model === undefined) {
     try {
-      model = factory.create(json.type, uniqueId, terria);
+      model = factory.create(
+        isJsonString(json.type) ? json.type : undefined,
```

Comment:
I think it's clearer to put this ternary outside factory.create

Reply:
done!